radio/src/mixer.cpp

```diff
@@ -851,25 +854,62 @@ void evalFlightModeMixes(uint8_t mode, uint8_t tick10ms)
 #endif
 
       //========== WEIGHT ===============
+      int16_t weight = GET_GVAR(MD_WEIGHT(md), GV_RANGELARGE_NEG, GV_RANGELARGE, mixerCurrentFlightMode);
+      weight = calc100to256_16Bits(weight);
       int32_t dv = (int32_t) v * weight;
+      int32_t dtrim = (int32_t) trim * weight;    
 
-      //========== OFFSET / AFTER ===============
-      if (apply_offset_and_curve) {
-        int16_t offset = GET_GVAR(MD_OFFSET(md), GV_RANGELARGE_NEG, GV_RANGELARGE, mixerCurrentFlightMode);
-        if (offset) dv += int32_t(calc100toRESX_16Bits(offset)) << 8;
+      //========== OFFSET AFTER ========= 
+      if (apply_offset_and_curve) { 
+        int16_t offset = GET_GVAR(MD_OFFSET(md), GV_RANGELARGE_NEG, GV_RANGELARGE, mixerCurrentFlightMode); 
+        if (offset) dv += int32_t(calc100toRESX_16Bits(offset)) << 8; 
       }
 
       //========== DIFFERENTIAL =========
 #if !defined(XCURVES)
       if (md->curveMode == MODE_DIFFERENTIAL) {
+        //Revert trim    
+        if (apply_offset_and_curve) {
+          if (!(mode & e_perout_mode_notrims)) {
+#if defined(VIRTUALINPUTS)
+            if (md->carryTrim == 0) {
+              dv -= dtrim;  
```

Comment:
3) Differential would also play on weight (and obviously this code should be executed before applying weight to the input)